packages/veritone-react-common/src/components/DataTable/index.js

```diff
@@ -0,0 +1,441 @@
+  TableHead as MuiTableHead,
+  TableRow,
+  TableCell
+} from 'material-ui/Table';
+import Paper from 'material-ui/Paper';
+
+import { injectInto } from 'helpers/react';
+import {
+  bool,
+  func,
+  number,
+  node,
+  objectOf,
+  any,
+  string,
+  oneOfType,
+  oneOf,
+  arrayOf
+} from 'prop-types';
+
+import withPagination from './withPagination';
+import withBasicBehavior from './withBasicBehavior';
+import styles from './styles/index.scss';
+
+
+/*
+ * BASE TABLE
+ */
+class _Table extends React.Component {
+  static propTypes = {
+    rowGetter: func.isRequired,
+    rowCount: number.isRequired,
+    rowRange: arrayOf(number),
+    rowHeight: number,
+    width: number,
+    footerElement: node,
+    focusedRow: number,
+    renderFocusedRowDetails: func,
+    onShowCellRange: func,
+    emptyRenderer: func,
+    children: node, // children should be Columns
+    showHeader: bool
+  };
+
+  static defaultProps = {
+    rowHeight: 75,
+    onShowCellRange: noop,
+    emptyRenderer: noop,
+    showHeader: true
+  };
+
+  render() {
+    const restProps = omit(this.props, [
+      'onShowCellRange',
+      'focusedRow',
+      'renderFocusedRowDetails'
+    ]);
+
+    return (
+      <CSSTransitionGroup
+        transitionEnter
+        transitionLeave={false}
+        transitionEnterTimeout={200}
+        transitionLeaveTimeout={200}
+        transitionName={{
+          enter: styles['enter'],
+          enterActive: styles['enterActive'],
+          leave: styles['leave'],
+          leaveActive: styles['leaveActive']
+        }}
+        component="div"
+      >
+        <Paper elevation={1} className={styles.table}>
+          {isNumber(this.props.focusedRow) && this.props.rowCount > 0
+            ? <SplitTableContainer
+                {...restProps}
+                focusedRow={this.props.focusedRow}
+                renderFocusedRowDetails={this.props.renderFocusedRowDetails}
+                key="split"
+              />
+            : <NormalTableContainer {...restProps} key="normal" />}
+        </Paper>
+      </CSSTransitionGroup>
+    );
+  }
+}
+
+/*
+ * Table body when no row is focused
+ */
+
+const NormalTableContainer = ({
+  rowCount,
+  rowRange,
+  footerElement,
+  children,
+  emptyRenderer,
+  showHeader,
+  selectable,
+  ...rest
+}) => {
+  return (
+    <div className={styles['table-wrapper']}>
+      <MuiTable>
+        {
+          showHeader &&
+          <TableHead
+            rowCount={rowCount}
+          >
+            {children}
+          </TableHead>
+        }
+        {
+          rowCount === 0
+          ? <MuiTableBody>
+              <TableRow>
+                <TableCell colSpan={children.length}>
+                  {emptyRenderer()}
+                </TableCell>
+              </TableRow>
+            </MuiTableBody>
+          : <TableBody 
+              {...rest}
+              rowRangeStart={(rowRange && rowRange[0])}
+              rowRangeEnd={(rowRange && rowRange[1]) || rowCount}
```

Comment:
I think this rowRange concept is a bit unnecessary -- i'm wondering why you introduced it. The idea was that the base/normal table has no concept of rendering anything other than 0->rowcount. Then the pagination and split-table layers translate that based on their own internal logic. Keeps the responsibilities completely separate.

Did you gain something from changing away from that?

Reply:
Yeah i observed that `rowCount` was being changed internally such that it's value was essentially rowsPerPage - 1. This behavior seemed odd since when i paged through a table and clicked on a row, i expected the row number to represent the actual position of that row in the dataset instead of the position of the row in the current view. This approach kept consistency with the value supplied through the `rowCount` prop and removed the need for any translation methods.

Reply:
ok no worries. i'll revert my changes back